apps/console/src/features/users/components/add-user-updated.tsx

```diff
@@ -0,0 +1,947 @@
+/**
+ * Copyright (c) 2021, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Field, FormValue, Forms, Validation } from "@wso2is/forms";
+import { Button, Hint, PasswordValidation, Popup } from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import React, { MutableRefObject, ReactElement, useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Dropdown, DropdownItemProps, DropdownProps, Form, Grid, Message, Radio } from "semantic-ui-react";
+import { SharedUserStoreConstants } from "../../core/constants";
+import { EventPublisher, SharedUserStoreUtils } from "../../core/utils";
+import { getUsersList } from "../../users/api/users";
+import { 
+    BasicUserDetailsInterface,
+    UserListInterface
+} from "../../users/models/user";
+import {
+    generatePassword, 
+    getConfiguration, 
+    getUsernameConfiguration
+} from "../../users/utils";
+import { PRIMARY_USERSTORE, USERSTORE_REGEX_PROPERTIES } from "../../userstores/constants/user-store-constants";
+import { ValidationDataInterface, ValidationFormInterface } from "../../validation/models";
+import { UserManagementConstants } from "../constants";
+
+/**
+ * Proptypes for the add user component.
+ */
+export interface AddUserProps {
+    initialValues: any;
+    triggerSubmit: boolean;
+    emailVerificationEnabled: boolean;
+    onSubmit: (values: BasicUserDetailsInterface) => void;
+    hiddenFields?: ("userStore" |"userName" | "firstName" | "lastName" | "password" | "email")[];
+    requestedPasswordOption?: "ask-password" | "create-password";
```

Comment:
Shall we create an enum for this?

Reply:
Fixed with fd5839362baf8e4e50397013a20cfa8b0b15fa4e